Use developer-friendly and precise technical language in docs (#22382)
gravitational/teleport: docs/pages/index.mdx

Teleport is an identity-native infrastructure access platform. With Teleport you can:

- Replace a mix of vaults, passwords, API keys and tokens with short-lived SSH and X.509 certs.

> - Have a single access point and Single-Sign-On for all your infrastructure - SSH, Kubernetes, databases, desktops, web applications, and more.

Reviewer: ```suggestion
- Have a single access point and Single-Sign-On for all of your infrastructure - SSH servers, Kubernetes clusters, databases, desktops, web applications, and more.
```

> - Write policies with Terraform or K8s CRDs for all clouds, environments and protocols and apply them with PR flow.

Reviewer: ```suggestion
- Write policies with Terraform or Kubernetes resources for all clouds, environments and protocols and apply them with PR flow.
```

> - Write policies with Terraform or K8s CRDs for all clouds, environments and protocols and apply them with PR flow.

Author: replaced with GitOps which is more precise term used in the industry

> - Record SSH and `kube exec` sessions, DB queries, Desktop recordings, web services and apps.

Reviewer 2: ```suggestion
- Record SSH and `kube exec` sessions, DB queries, Windows desktop sessions, and application access.
```

Author: applied with a couple of tweaks

> - Record SSH and `kube exec` sessions, DB queries, Desktop recordings, web services and apps.

Reviewer: - I would say `kubectl exec` if you're going to put it in backtick. 
- We say "record desktop recordings" which feels a bit repetitive.

Author: done
